### PR TITLE
fix: Show actual execution data for production executions even if pin data exists

### PR DESCRIPTION
**Pull request by alexgrozav**

### CODE REVIEW

File: packages/editor-ui/src/mixins/__tests__/workflowHelpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+				workflows: useWorkflowsStore(),
+			};
+		});
+
+		it('should return true if no active execution is set', () => {
+			const result = shouldReplaceInputDataWithPinData();
+			expect(result).toBe(true);
+		});
+
+		it('should return true if active execution is set and mode is manual', async () => {
```
Comment (ivov) [inline packages/editor-ui/src/mixins/__tests__/workflowHelpers.test.ts:23]:
```suggestion
		it('should return true if active execution is set and mode is manual', () => {
```

File: packages/editor-ui/src/mixins/__tests__/workflowHelpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			pinia.state.value.workflows.activeWorkflowExecution = { mode: 'manual' };
+
+			const result = shouldReplaceInputDataWithPinData();
+			expect(result).toBe(true);
+		});
+
+		it('should return false if active execution is set and mode is not manual', async () => {
```
Comment (ivov) [inline packages/editor-ui/src/mixins/__tests__/workflowHelpers.test.ts:30]:
```suggestion
		it('should return false if active execution is set and mode is not manual', () => {
```

File: packages/editor-ui/src/mixins/workflowHelpers.ts [hunk 3]
```diff
@@ -223,32 +232,36 @@ function connectionInputData(
...
-			acc.push({
-				json: pinData[0],
-				pairedItem: {
-					item: index,
-					input: 1,
-				},
-			});
-		}
+	if (shouldReplaceInputDataWithPinData()) {
+		const parentPinData = parentNode.reduce((acc: INodeExecutionData[], parentNodeName, index) => {
```
Comment (ivov) [inline packages/editor-ui/src/mixins/workflowHelpers.ts:238]:
```suggestion
		const parentPinData = parentNode.reduce<INodeExecutionData[]>((acc, parentNodeName, index) => {
```

File: packages/editor-ui/src/views/NodeView.vue [hunk 4]
```diff
@@ -3604,6 +3604,8 @@ export default defineComponent({
 							type: 'error',
 						});
 					}
+				} else if (json && json.command === 'setActiveExecution') {
```
Comment (ivov) [inline packages/editor-ui/src/views/NodeView.vue:3607]:
```suggestion
				} else if (json?.command === 'setActiveExecution') {
```

File: packages/editor-ui/src/mixins/workflowHelpers.ts [hunk 5]
```diff
@@ -189,6 +189,15 @@ function getNodeTypes(): INodeTypes {
...
 }
 
+export function shouldReplaceInputDataWithPinData() {
+	const workflowsStore = useWorkflowsStore();
+	return (
+		!workflowsStore.activeWorkflowExecution ||
+		(workflowsStore.activeWorkflowExecution &&
+			workflowsStore.activeWorkflowExecution.mode === 'manual')
+	);
+}
```
Comment (ivov) [inline packages/editor-ui/src/mixins/workflowHelpers.ts:199]: Or shorter:

```ts
export function shouldReplaceInputDataWithPinData() {
	const { activeWorkflowExecution } = useWorkflowsStore();
	return !activeWorkflowExecution || activeWorkflowExecution?.mode === 'manual';
}
```

Comment (ivov) [reply packages/editor-ui/src/mixins/workflowHelpers.ts:199]: We could make this a getter in the workflows store? To avoid adding to mixins.

**Review by ivov — DISMISSED** [review]
Confirmed working!

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.232.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.232.0)

**Pull request merged**